crd/multitenancy/api/v1alpha1/utils.go

```diff
@@ -0,0 +1,30 @@
+package v1alpha1
+
+// IsReady checks if all the required fields in the MTPNC status are populated
+func (m *MultitenantPodNetworkConfig) IsReady() bool {
+	if m.Status.PrimaryIP == "" ||
+		m.Status.MacAddress == "" ||
+		m.Status.NCID == "" ||
+		m.Status.GatewayIP == "" {
+		return false
+	}
+
+	// Check if InterfaceInfos slice is not empty
+	if len(m.Status.InterfaceInfos) == 0 {
+		return false
+	}
+
+	// Check if each InterfaceInfo has all required fields populated
+	for _, interfaceInfo := range m.Status.InterfaceInfos {
```

Comment:
line 13, we check whether length of interfaceInfos object is empty, do we need to check if all fields are empty again?

Reply:
If the interfaceInfos array is not empty but some fields are not populated in the InterfaceInfos, I believe that means the MTPNC is not ready.